dbschema/src/main/java/org/folio/rest/persist/ddlgen/Index.java

```diff
@@ -83,24 +84,45 @@ public String getFinalSqlExpression(String tableLoc) {
           result .append(" , ");
         }
         String [] rawExpandedTerm = splitIndex[i].split("\\.");
-        StringBuilder expandedTerm = formatExpandedTerm(rawExpandedTerm);
-        result.append(tableLoc).append(".jsonb").append(expandedTerm);
+        StringBuilder expandedTerm = formatExpandedTerm(tableLoc + ".jsonb",rawExpandedTerm);
+        result.append(expandedTerm);
       }
       result.append(")");
       return result.toString();
     }
 
   }
-  private StringBuilder formatExpandedTerm(String[] rawExpandedTerm) {
-    StringBuilder expandedTerm = new StringBuilder();
 
+  private StringBuilder formatExpandedTerm(String table, String[] rawExpandedTerm) {
```

Comment:
Can you add unit tests for this method? The most easy way is using https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests , for example:

    @ParameterizedTest
    @CsvSource({
      "a    , ->>a",
      "a.b  , ->'a'->>'b'",
      "a.b.c, ->'a'->'b'->>'c'",
    })
    void appendExpandedSimpleTerm(String term, String expectedSql) {
      StringBuilder result = new StringBuilder();
      Index.appendExpandedTerm(result, "table", term);
      assertThat(result.toString(), is("table.jsonb" + expecedSql));
    }

The same for array terms.